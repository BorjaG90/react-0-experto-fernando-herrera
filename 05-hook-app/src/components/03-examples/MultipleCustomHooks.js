import React from 'react';
import {useFetch} from '../../hooks/useFetch';

import './effects.css';

export const MultipleCustomHooks = () => {

  useFetch(`https://www.breakingbadapi.com/api/quotes/1`);
  return (
    <div>
      <h1>Custom Hooks!!!</h1>
    </div>
  )
}
