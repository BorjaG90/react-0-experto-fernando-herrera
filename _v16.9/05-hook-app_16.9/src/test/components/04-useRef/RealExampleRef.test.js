import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';

describe('pruebas en <RealExampleRef />', () => { 
  const wrapper = shallow(<RealExampleRef />);

  test('should mostrarse correctamente', () => { 
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });

  test('should de mostrar el componente <MultipleCustomHook />', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  });
})