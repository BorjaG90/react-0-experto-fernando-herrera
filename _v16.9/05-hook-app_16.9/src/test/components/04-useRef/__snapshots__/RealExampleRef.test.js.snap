// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <RealExampleRef /> should mostrarse correctamente 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
